circleci/client.go

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2019 Jobteaser
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package circleci expose function to manage environment variable in
+// CircleCI project.
+package circleci // import "github.com/jobteaser-oss/circleci-env"
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// Client contains anm `http.Client` and a URL.
+type Client struct {
+	http *http.Client
+	url  *url.URL
+}
+
+// NewClient returns a pointer of `Client` struct with timeout and
+// authentication properly configured.
+func NewClient(token string) (*Client, error) {
+	const timeout = 5 * time.Second
+
+	dialer := net.Dialer{Timeout: timeout}
+
+	transport := http.Transport{
+		Dial:                dialer.Dial,
+		TLSHandshakeTimeout: timeout,
+	}
+
+	client := http.Client{
+		Transport: &transport,
+		Timeout:   timeout,
+	}
+
+	uri, err := url.Parse("https://circleci.com/api/v1.1")
+	if err != nil {
```

Comment:
Since the URL is not an input, this "should not happen", so a panic would be more appropriate to handle this error